engine/src/test/java/org/camunda/bpm/engine/test/api/mgmt/LicenseUserOperationLogTest.java

```diff
@@ -0,0 +1,140 @@
+public class LicenseUserOperationLogTest {
+
+  private final static String LICENSE_KEY = "testLicenseKey";
+  private static final String USER_ID = "testUserId";
+
+  @Rule
+  public ProcessEngineRule engineRule = new ProvidedProcessEngineRule();
+
+  ProcessEngine processEngine;
+  ManagementService managementService;
+  HistoryService historyService;
+  IdentityService identityService;
+
+  @Before
+  public void init() {
+    processEngine = engineRule.getProcessEngine();
+    managementService = processEngine.getManagementService();
+    historyService = processEngine.getHistoryService();
+    identityService = processEngine.getIdentityService();
+  }
+
+  @After
+  public void tearDown() {
+    managementService.deleteLicenseKey();
+    identityService.clearAuthentication();
+  }
+
+  @Test
+  public void shouldUpdateOperationLogWhenSetLicense() {
+    // given
+    identityService.setAuthenticatedUserId(USER_ID);
+
+    // when
+    managementService.setLicenseKey(LICENSE_KEY);
+    UserOperationLogEntry entry = historyService.createUserOperationLogQuery().singleResult();
+
+    // then
+    assertOperationLogEntry(entry, UserOperationLogEntry.OPERATION_TYPE_CREATE);
+  }
+
+  @Test
+  public void shouldUpdateOperationLogWhenUpdateLicense() {
+    // given
+    managementService.setLicenseKey("oldLicense");
+    identityService.setAuthenticatedUserId(USER_ID);
+
+    // when
+    managementService.setLicenseKey(LICENSE_KEY);
+    UserOperationLogEntry entry = historyService.createUserOperationLogQuery().singleResult();
+
+    // then
+    assertOperationLogEntry(entry, UserOperationLogEntry.OPERATION_TYPE_UPDATE);
+  }
+
+  @Test
+  public void shouldUpdateOperationLogWhenUpdateLicenseDuplicate() {
+    // given
+    managementService.setLicenseKey(LICENSE_KEY);
+    identityService.setAuthenticatedUserId(USER_ID);
+    
+    // when
+    managementService.setLicenseKey(LICENSE_KEY);
+    UserOperationLogEntry entry = historyService.createUserOperationLogQuery().singleResult();
+    
+    // then
+    assertOperationLogEntry(entry, UserOperationLogEntry.OPERATION_TYPE_UPDATE);
+  }
+
+  @Test
+  public void shouldUpdateOperationLogWhenDeleteLicense() {
+    // given
+    managementService.setLicenseKey(LICENSE_KEY);
+    identityService.setAuthenticatedUserId(USER_ID);
+
+    // when
+    managementService.deleteLicenseKey();
+    UserOperationLogEntry entry = historyService.createUserOperationLogQuery().singleResult();
+
+    // then
+    assertOperationLogEntry(entry, UserOperationLogEntry.OPERATION_TYPE_DELETE);
+  }
+
+  @Test
+  public void shouldNotUpdateOperationLogWhenDeleteNonExistingLicense() {
+    // given
+    identityService.setAuthenticatedUserId(USER_ID);
+    
+    // when
+    managementService.deleteLicenseKey();
+    UserOperationLogEntry entry = historyService.createUserOperationLogQuery().singleResult();
+    
+    // then
+    assertThat(entry).isNull();
+  }
+
```

Comment:
Which operation logs do we get when we update a license key that was set through properties API with the new API?